src/Migrations/M042FixPostsListMigration.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+// phpcs:disable Generic.Files.LineLength.MaxExceeded
+
+namespace P4\MasterTheme\Migrations;
+
+use P4\MasterTheme\MigrationRecord;
+use P4\MasterTheme\MigrationScript;
+
+/**
+ * Migrate Posts List blocks with errors to Posts List blocks fix.
+ */
+class M042FixPostsListMigration extends MigrationScript
+{
+    /**
+     * Perform the actual migration.
+     *
+     * @param MigrationRecord $record Information on the execution, can be used to add logs.
+     * phpcs:disable SlevomatCodingStandard.Functions.UnusedParameter -- interface implementation
+     */
+    public static function execute(MigrationRecord $record): void
+    {
+        $check_is_valid_block = function ($block) {
+            return self::check_is_valid_block($block);
+        };
+
+        $transform_block = function ($block) {
+            return self::transform_block($block);
+        };
+
+        Utils\Functions::execute_block_migration(
+            Utils\Constants::BLOCK_QUERY,
+            $check_is_valid_block,
+            $transform_block,
+        );
+    }
+    // phpcs:enable SlevomatCodingStandard.Functions.UnusedParameter
+
+    /**
+     * Check whether a block is a Query Loop block.
+     *
+     * @param array $block - A block data array.
+     */
+    private static function check_is_valid_block(array $block): bool
+    {
+        // Check if the block is valid.
+        if (!is_array($block)) {
+            return false;
+        }
+
+        // Check if the block has a 'blockName' key.
+        if (!isset($block['blockName'])) {
+            return false;
+        }
+
+        // Check if the block is a Query Loop block.
+        return $block['blockName'] === Utils\Constants::BLOCK_QUERY;
+    }
+
+    /**
+     * Update the post title, post terms and paragraph of the
+     * Posts List block
+     *
+     * @param array $block - The current posts list block.
+     * @return array - array of blocks.
+     */
+    private static function transform_block(array $block): array
+    {
+        if ($block['attrs']['namespace'] === Utils\Constants::BLOCK_POSTS_LIST) {
```

Comment:
I would add this in the `check_is_valid_block` function instead since we don't want to migrate otherwise